src/warehouse/index.js

```diff
@@ -208,6 +208,23 @@ function setDataFromInputAndComputeColumnTypes(
   level = 0,
 ) {
   if (!input || !isObject(input)) return;
+  if (
+    (completePrefix.endsWith('context_traits_') || completePrefix === 'group_traits_') &&
+    isStringLikeObject(input)
+  ) {
+    if (prefix === 'context_traits_') {
+      appendColumnNameAndType(
+        utils,
+        eventType,
+        `${prefix.length > 0 ? prefix : completePrefix}`,
```

Comment:
We are already checking for prefix === 'context_traits_'. `prefix.length` will always be greater than 0.

Reply:
Done